### ISSUE TITLE
Update extension tutorial

### ISSUE THREAD

**Issue by jasongrout** [jupyterlab/jupyterlab#6568]
I just walked through the xkcd tutorial with a group. A couple of quick notes to self:

1. @kellyrowland brought up that some xkcd comics are NSFW or otherwise insensitive or inappropriate (thanks! I totally agree!). Perhaps we should change it to the astronomy picture of the day from NASA?
2. We should change things to use a MainAreaWidget
3. Even with my updates, there are still some places where the `addToMainArea` should be replaced with `add(widget, 'main')`

**Comment by jasongrout** [comment]
I've made a lot of progress on this at https://github.com/jupyterlab/jupyterlab_apod

- [ ] Use MainAreaWidget as a first-class thing (i.e., set its id, etc., so the restore works - treat APODWidget *just* as content)
- [ ] Update actual tutorial
- [ ] Delete unnecessary coreutils dependency (since we aren't using JSONExt anymore)

**Comment by kellyrowland** [comment]
Thanks very much for putting this up. :slightly_smiling_face:

**Comment by jasongrout** [comment]
Thanks again for helping us be aware. I realized I didn't give you credit above for your comment, so edited the description.

**Comment by jasongrout** [comment]
I played with changing the example so that it pulled up only public domain images, with something like:

```typescript
    let data: APODResponse;
    let attempts = 20;

    // Fetch info about a random picture until we find a public domain image.
    do {
      const response = await fetch(`https://api.nasa.gov/planetary/apod?api_key=DEMO_KEY&date=${this.randomDate()}`);
      data = await response.json() as APODResponse;
      attempts -= 1
    } while (attempts > 0 && (data.copyright || data.media_type !== 'image'))
    
    if (attempts === 0) {
      this.header.innerText = 'Could not find a public domain APOD picture'
    } else {
      // Populate the image
      this.img.src = data.url;
      this.img.title = data.title;
      this.header.innerText = data.title;
    }
```

However, this (a) complicates the code quite a bit, (b) makes it so that it is really easy to run over the demo limits, and (c) I noticed at least one picture with a copyright watermark that wasn't noted in the data as having copyright, so it's not completely accurate. That said, I see it often requesting several images, so it seems pretty likely in general to have copyrighted content coming up.

I think having copyrighted images fetched may be an issue if there is a recording of this tutorial that is a commercial product. There might be some allowance here as fair use (even in commercial contexts?). I think it won't be an issue generally.

I'll leave the code as-is, retrieving images that may be copyrighted for the reasons above. If you need to make sure the images are not copyrighted, perhaps we can leave that as an exercise for the reader?

**Comment by blink1073** [comment]
Fixed in #6656.  Thanks @jaipreet-s  and @jasongrout!
